src/evo/deterministicmns.cpp

```diff
@@ -1596,8 +1624,21 @@ bool CheckProUpRevTx(CDeterministicMNManager& dmnman, const CTransaction& tx, gs
 
     auto mnList = dmnman.GetListForBlock(pindexPrev);
     auto dmn = mnList.GetMN(opt_ptx->proTxHash);
-    if (!dmn)
+    if (!dmn) {
         return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-protx-hash");
+    }
+
+    const bool is_v23_active{DeploymentActiveAfter(pindexPrev, Params().GetConsensus(), Consensus::DEPLOYMENT_V23)};
+
+    // Don't allow legacy scheme versioned transactions after upgrading to basic scheme
+    if (is_v23_active && dmn->pdmnState->nVersion >= ProTxVersion::BasicBLS && opt_ptx->nVersion == ProTxVersion::LegacyBLS) {
+        return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-protx-version-downgrade");
+    }
+
+    // Nodes using the legacy scheme must first upgrade to the basic scheme before upgrading further
+    if (is_v23_active && dmn->pdmnState->nVersion == ProTxVersion::LegacyBLS && opt_ptx->nVersion > ProTxVersion::BasicBLS) {
+        return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-protx-version-upgrade");
+    }
```

Comment:
Revoking a MN (`ProUpRevTx`) doesn't change its `nVersion`, you don't need this logic here imo. Could maybe enforce that versions match instead i.e.
`if (is_v23_active && dmn->pdmnState->nVersion != opt_ptx->nVersion) { return state.Invalid(...); }`.

Reply:
Since this is changing the behavior of `protx revoke`, the whole set of ProTx version restrictions have been spun off into [dash#6729](https://github.com/dashpay/dash/pull/6729).